src/Features/CSharp/Portable/ExtractMethod/CSharpExtractMethodService.cs

```diff
@@ -16,16 +16,13 @@ namespace Microsoft.CodeAnalysis.CSharp.ExtractMethod;
 [method: ImportingConstructor]
 [method: SuppressMessage("RoslynDiagnosticsReliability", "RS0033:Importing constructor should be [Obsolete]", Justification = "Used in test code: https://github.com/dotnet/roslyn/issues/42814")]
 internal sealed partial class CSharpExtractMethodService() : AbstractExtractMethodService<
-    CSharpExtractMethodService.CSharpSelectionValidator,
-    CSharpExtractMethodService.CSharpMethodExtractor,
-    CSharpExtractMethodService.CSharpSelectionResult,
```

Comment:
got rid of theese generics.  they are not needed, and just add type system complexity.